File: convert jpg to png.java
import java.awt.image.BufferedImage;
import java.io.File;
import javax.imageio.ImageIO;

public class JpgToPngConverter {
    public static void main(String[] args) {
        String jpgFilename = "input.jpg";
        String pngFilename = "output.png";

        try {
            // Read the input JPG image
            BufferedImage image = ImageIO.read(new File(jpgFilename));

            // Write the output PNG image
            File output = new File(pngFilename);
            ImageIO.write(image, "png", output);

            System.out.println("Conversion complete!");
        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}

File: convert jpg to png.rb
require 'chunky_png'
require 'mini_magick'

def convert_jpg_to_png(jpg_filename, png_filename)
  # Read the JPG image data
  image = MiniMagick::Image.open(jpg_filename)

  # Convert to PNG format using ChunkyPNG
  png_data = ChunkyPNG::Image.from_io(StringIO.new(image.to_blob)) 

  # Write the PNG data to a file
  File.open(png_filename, 'wb') do |file|
    png_data.write(file)
  end
end
